packages/language-server/src/utils.ts

```diff
@@ -43,8 +57,21 @@ export function getLastPartOfPath(path: string): string {
     return path.replace(/\\/g, '/').split('/').pop() || '';
 }
 
-export function flatten<T>(arr: T[][]): T[] {
-    return arr.reduce((all, item) => [...all, ...item], []);
+export function flatten<T>(arr: (T | T[])[]): T[] {
```

Comment:
Expanded to make the behaviour closer to native JS `flatMap`: flatten arrays, keep non array values as is. The adaptation was needed to spare the client code in the function I introduced to do this dance itself.

Existing tests say this is a backward compatible change :innocent: